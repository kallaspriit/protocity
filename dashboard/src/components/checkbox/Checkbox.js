import React, { PropTypes } from 'react';
import classNames from 'classnames';
// import './checkbox.scss';

function Checkbox({ className, onChange, checked, defaultChecked, name, disabled, children }) {
	const wrapClassName = classNames('checkbox', className);

	return (
		<label className={wrapClassName}>
			<input
				className="checkbox__input"
				type="checkbox"
				name={name}
				onChange={onChange}
				checked={checked}
				defaultChecked={defaultChecked}
				disabled={disabled}
			/>
			{children}
		</label>
	);
}

Checkbox.propTypes = {
	children: PropTypes.node,
	className: PropTypes.string,
	name: PropTypes.string,
	checked: PropTypes.bool,
	defaultChecked: PropTypes.bool,
	onChange: PropTypes.func,
	disabled: PropTypes.bool,
};

Checkbox.defaultProps = {
	disabled: false,
};

export default Checkbox;