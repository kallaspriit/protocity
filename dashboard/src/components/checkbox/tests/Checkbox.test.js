import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Checkbox from '../Checkbox';

const onChangeMock = jest.fn();

const props = {
	className: 'test-class',
	name: 'Checkbox',
	checked: false,
	onChange: onChangeMock,
	disabled: false,
};

const component = mount(
	<Checkbox {...props}>
		<div>children</div>
	</Checkbox>,
);

describe('Checkbox', () => {
	it('have correct props', () => {
		component.setProps({
			className: 'new-test-class',
			name: 'New Checkbox',
			checked: !props.checked,
			disabled: !props.disabled,
		});

		expect(toJson(component)).toMatchSnapshot();
	});
});