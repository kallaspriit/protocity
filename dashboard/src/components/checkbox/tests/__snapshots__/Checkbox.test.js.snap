exports[`Checkbox have correct props 1`] = `
<Checkbox
  checked={true}
  className="new-test-class"
  disabled={true}
  name="New Checkbox"
  onChange={[Function]}>
  <label
    className="checkbox new-test-class">
    <input
      checked={true}
      className="checkbox__input"
      disabled={true}
      name="New Checkbox"
      onChange={[Function]}
      type="checkbox" />
    <div>
      children
    </div>
  </label>
</Checkbox>
`;
